### PR TITLE
Test dev examples

### PR DESCRIPTION
**Pull request by brisvag**
# References and relevant issues
Alternative to #6763.

# Description
Test dev examples to ensure they don't go out of sync.

### CODE REVIEW

**Comment by brisvag** [comment]
@Czaki the failing example is related to 2d hash in labels... is it even relevant anymore? If so, could you fix it?

**Comment by Czaki** [comment]
No. We have changed the hashing mechanism. I will try to check it tomorrow.

**Comment by Czaki** [comment]
The current mechanism is totally different, so I think that whole example could be removed.

File: examples/dev/demo_shape_creation.py [hunk 1]
```diff
@@ -34,77 +34,77 @@ def time_me(label, func):
...
+    help='concatenate all coordinates to a single mesh',
+)
+parser.add_argument(
+    '-v', '--view', action='store_true', help='show napari viewer'
+)
+parser.add_argument(
+    '--properties', action='store_true', help='add dummy shape properties'
+)
+
+args = parser.parse_args()
```
Comment (Czaki) [inline examples/dev/demo_shape_creation.py:66]: This is bad change. As it trigger always parsing content of `sys.argv` and may lead to crash of test. I think that we should use `init_globals` argument of `run_path` to detect such use. maybe something like:
```suggestion
if locals().get("test_run", False):
    args = parser.parse_args([])
else:
    args = parser.parse_args([])
```

File: napari/_tests/test_examples.py [hunk 2]
```diff
@@ -69,12 +71,31 @@ def raise_errors(etype, value, tb):
 
     monkeypatch.setattr(notification_manager, 'receive_error', raise_errors)
 
-    # run the example!
+
+def _run_example(example_path):
     try:
-        runpy.run_path(str(EXAMPLE_DIR / fname))
+        runpy.run_path(example_path)
```
Comment (Czaki) [inline napari/_tests/test_examples.py:77]:
```suggestion
        runpy.run_path(example_path, init_globals={"test_run", True})
```

**Comment by brisvag** [comment]
I don't liek the idea of having test-specific code in the examples... I'd rather just skip this one or refactor it ^^'

**Comment by Czaki** [comment]
> I don't liek the idea of having test-specific code in the examples... I'd rather just skip this one or refactor it ^^'

Hm. Looking to documentation, maybe we should just monkeypatch `sys.argv`?

> A number of alterations are also made to the [sys](https://docs.python.org/3/library/sys.html#module-sys) module. Firstly, [sys.path](https://docs.python.org/3/library/sys.html#sys.path) may be altered as described above. sys.argv[0] is updated with the value of path_name and sys.modules[__name__] is updated with a temporary module object for the module being executed. All modifications to items in [sys](https://docs.python.org/3/library/sys.html#module-sys) are reverted before the function returns.

**Comment by brisvag** [comment]
What do you think of this version?

**Review by Czaki — APPROVED** [review]

File: examples/dev/issue-6456.py [hunk 3]
```diff
@@ -19,4 +21,5 @@
         )
 
 if __name__ == '__main__':
-    napari.run()
+    if 'PYTEST_CURRENT_TEST' not in os.environ:
```
Comment (psobolewskiPhD) [inline examples/dev/issue-6456.py]: is this needed or leftover from my PR?

**Review by psobolewskiPhD — APPROVED** [review]
just that one small question

File: examples/dev/issue-6456.py (hunk 3 above)
Comment (brisvag) [reply examples/dev/issue-6456.py]: leftover indeed!

**Comment by brisvag** [comment]
Fail seems unrelated.

**Pull request merged**